hopr/hopr-lib/src/lib.rs

```diff
@@ -678,7 +639,10 @@ impl Hopr {
         self.cfg.chain.announce
     }
 
-    pub async fn run(&self) -> errors::Result<(HoprSocket, HashMap<HoprLibProcesses, JoinHandle<()>>)> {
+    pub async fn run<#[cfg(feature = "session-server")] T: HoprSessionServerActionable + Clone + Send + 'static>(
```

Comment:
Isn't it better to have two separate prototypes, that cannot be feature-enabled at the same time?
It would be more readable and AFAIK there's no major difference on the callsite than this.

Reply:
Maybe yes. Will revisit once the changes are complete.